src/function/list/list_prepend_function.cpp

```diff
@@ -28,21 +30,71 @@ struct ListPrepend {
     }
 };
 
+static void resolveEmptyList(const ScalarBindFuncInput& input, std::vector<LogicalType>& types) {
```

Comment:
Let's refactor the `resolveEmptyList` in the `list_prepend` and `list_append` into a common file. So we don't duplicate code in `list_append` and `list_prepend`